lib/java_buildpack/framework/sealights_agent.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/framework'
+require 'java_buildpack/component/versioned_dependency_component'
+require 'fileutils'
+
+module JavaBuildpack
+  module Framework
+
+    # Encapsulates the functionality for enabling zero-touch Sealights support.
+    class SealightsAgent < JavaBuildpack::Component::VersionedDependencyComponent
+      # include JavaBuildpack::Util
+
+      def initialize(context)
+        super(context)
+        @logger = JavaBuildpack::Logging::LoggerFactory.instance.get_logger SealightsAgent
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        download_zip(false)
+        @droplet.copy_resources
```

Comment:
Is this something you envision being useful? I don't see any [resources being added as part of your commit](https://github.com/cloudfoundry/java-buildpack/tree/main/resources). If you don't have any resources to copy, you can likely remove this call.